File: problem/gcd/6/main.lua
-- main.lua: main file for "gcd" solutions
--
-- Copyright (C) 2020-2021  Rudy Matela
-- All rights reserved.
--
--
-- This file is part of the CScx Online Judge.
--
-- This is appended to the submitted Lua program and tests the triple function.
--
-- The submitted file processes the standard input and this processes the
-- "in.txt" file.
io.input("in.txt")
a, b = io.read("*n", "*n")
while a and b do
  print(gcd(a,b))
  a, b = io.read("*n", "*n")
end

File: problem/gcd/6/main.rb

# main.rb: main file for solutions to the gcd problem
#
# This is appended to the submitted Ruby program and tests the gcd function.
#
# The submitted file processes the standard input and this processes the
# "in.txt" file.
#
#
# Copyright (C) 2020-2021  Rudy Matela
# All rights reserved.
#
#
# This file is part of the CSCX Online Judge.
#
# This is to be appended to the submitted file.

# standard input is already processed
# as the submitted programmed is run when imported
File.foreach("in.txt") do |line|
  a, b = line.split.map { |x| x.to_i }
  puts gcd(a,b)
end
